### PR TITLE
Spark 2.4

### PR DESCRIPTION
**Pull request by elmiko**
This change brings in support for Spark 2.4.0, it also updates the versioning on the incomplete images.

### CODE REVIEW

**Comment by elmiko** [comment]
@tmckayus @crobby @Jiri-Kremser ptal

File: docs/spark-version-update-process.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+## prerequisites
+
+* shell access
+* an editor available
+* access to the `docker` command line tool and a registry (for testing)
+* [cekit](https://cekit.readthedocs.io/en/latest/) available
+
+## procedure
+
+1. update version and download link in `image.yaml`
```
Comment (tmckayus) [inline docs/spark-version-update-process.md]: reference the change-yaml.sh utility script here?

Comment (elmiko) [reply docs/spark-version-update-process.md]: yes, this was just an oversight on my part. i will go back and add the script references.

**Comment by elmiko** [comment]
one more thing i found while reading through the `change-yaml.sh` file, i need to update the md5 based logic in there to use the new sha512 stuff.

**Comment by elmiko** [comment]
fwiw, i made a request to the cekit project for a feature =)

https://github.com/cekit/cekit/issues/471

**Comment by tmckayus** [comment]
I think while we're waiting for the cekit feature, we can just do the extra download.

**Comment by elmiko** [comment]
ok, cool. i'll get that patch up

**Comment by elmiko** [comment]
added the sha512 changes to `change-yaml.sh`, just need to update the docs and we should be good to go.

**Comment by elmiko** [comment]
@tmckayus added more content about the script files, let me know what you think.

**Comment by Jiri-Kremser** [comment]
\o/

**Pull request merged**